### PR TITLE
Implemented the Data Service for Exercise class.

### PR DESCRIPTION
**Pull request by nicanorgarcia**
Database access for the Exercise class. Missing equal (==) method implementation.

### CODE REVIEW

File: app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDA.java [hunk 1]
```diff
@@ -2,64 +2,122 @@
...
     @NotNull
     private String exerciseName;
-    private String examplePath;
-    private String instructionsPath;
+    private String instructionTextPath;
+    private String instructionVideoPath;
+    private String fragmentClassName;
     private String exerciseType;
+    @Property
+    private String description;
```
Comment (AKuederle) [inline app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDA.java]: I think we should be coherent with the naming and call these two shortDescription and shortInstructions like in the dataclass

File: app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDataService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+public class ExerciseDataService {
+    private Context invocationcontext;
+    public ExerciseDataService(Context context){
+        this.invocationcontext = context;
+    }
+
+    public void saveExercise(Exercise exercise){
+        ExerciseDA tmp = new ExerciseDA(exercise);
+        DaoMaster.DevOpenHelper helper = new DaoMaster.DevOpenHelper(invocationcontext, "apkinsondb");
```
Comment (AKuederle) [inline app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDataService.java]: This is fine for now, but we should refactor this to remove redundant code

File: app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDataService.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public void deleteExercise(ExerciseDA exercise){
+        DaoMaster.DevOpenHelper helper = new DaoMaster.DevOpenHelper(invocationcontext, "apkinsondb");
+        Database db = helper.getWritableDb();
+        DaoSession session = new DaoMaster(db).newSession();
+        session.getExerciseDADao().delete(exercise);
+        db.close();
+    }
+    public Exercise getExercise(){
```
Comment (AKuederle) [inline app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDataService.java]: Shouldn't this function at least get a primary key to specify which excerse I want?

**Comment by AKuederle** [comment]
@nicanorgarcia can you rebase this branch onto the current develop to resolve the conflicts? And have a look at my comments. In particular the one about getExercises. All the other stuff is kind of optional

**Comment by AKuederle** [comment]
@nicanorgarcia Any updates on this one?

File: app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDataService.java (hunk 2 above)
Comment (nicanorgarcia) [reply app/src/main/java/com/sma2/sma2/DataAccess/ExerciseDataService.java]: We will have to refactor at some point. But for now, lets get things done quick and dirty

**Comment by nicanorgarcia** [comment]
I will delete this branch, had to many conflicts, started a new one

**Pull request closed without merging**